### PR TITLE
Add scale info document

### PR DESCRIPTION
**Pull request by p-schuler**
# PR for issue #1541 

Adding documentation for the scalability improvements and solution we are providing. This describes the solution as well as the challenge when dealing with multiple LNS instances. It also addresses the problem with different deduplication strategies.

### CODE REVIEW

File: docs/user-guide/deployment-scenarios.md [hunk 1]
```diff
@@ -30,16 +30,18 @@ There are multiple strategies that you can take to improve the availability of t
 IoT Hub and Redis need to be taken into consideration as well. There is documentation for all of these services explaining
 the different options for HA and scalability.
 
+With the version 2.1 we introduced a major improvement for multi gateway deployments that
```
Comment (danigian) [inline docs/user-guide/deployment-scenarios.md:33]:
```suggestion
With the version 2.1.0 we introduced a major improvement for multi gateway deployments that
```

File: docs/user-guide/deployment-scenarios.md [hunk 2]
```diff
@@ -30,16 +30,18 @@ There are multiple strategies that you can take to improve the availability of t
...
+allow the LNS to scale much better when using `DeduplicationMode.None` or
```
Comment (danigian) [inline docs/user-guide/deployment-scenarios.md:34]:
```suggestion
allows the LNS to scale much better when using `DeduplicationMode.None` or
```

File: docs/user-guide/scalability.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Scalability
+
+Different [deployment scenarios](./deployment-scenarios.md) have impact
+on the scalability of your LNS (LoRa network server).
```
Comment (danigian) [inline docs/user-guide/scalability.md:4]:
```suggestion
on the scalability of your LNS (LoRaWAN Network Server).
```

File: docs/user-guide/deployment-scenarios.md [hunk 4]
```diff
@@ -30,16 +30,18 @@ There are multiple strategies that you can take to improve the availability of t
...
+`DeduplicationMode.Drop`. All the deployment scenarios are now scaling to similiar levels
```
Comment (danigian) [inline docs/user-guide/deployment-scenarios.md:35]: Isn't the session affinity valid only for 'Drop'?

File: docs/user-guide/scalability.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+connection open/close pattern seriously affecting scalability of the LNS.
+
+Starting with version 2.1 we introduce a concept for connection ownership
+ensuring we only keep one active connection / LNS most of the time.
+
+To facilitate this, we track the last LNS that won the race to process
+a particular message. This LNS is then given an edge for future message
+processing for that device by a configurable amount. This allows
+the owning gateway to keep the connection open and keep processing messages
+without having to fight for the connection from other LNS's. LNS's that do not
```
Comment (danigian) [inline docs/user-guide/scalability.md:27]:
```suggestion
without having to fight for the connection from other LNSes. LNSes that do not
```

File: docs/user-guide/scalability.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+it won't get any new messages and won't notice, the ownership change
+unless it is being informed by the function.
+
+## Deduplication
+
+There is a feature built into the LNS to
+[deduplicate messages](./../adr/007_message_deduplication.md), to
+deal with duplicate messages on multiple LNS. This does
+have an effect on scalability. Higher scalability as described above
+can only be achieved with `None` and `Drop`. Any other settings
```
Comment (danigian) [inline docs/user-guide/scalability.md:51]:
```suggestion
can only be achieved with `Drop`. Any other settings
```
Same as above, I think we don't call deduplication part of the bundler with 'None' strategy

File: docs/user-guide/deployment-scenarios.md (hunk 4 above)
Comment (p-schuler) [reply docs/user-guide/deployment-scenarios.md:35]: absolutely.

**Review by danigian — APPROVED** [review]

**Pull request merged**